### PR TITLE
Add column endpoint_applications to list the installed applications in a computer. Closes #8

### LINKED ISSUE #8: Add support in `vanta_computer` to list the applications installed on a user's device
The UI shows a list of applications per computer/user installed on the device. Can we bring in the application inventory on the `vanta_computer` table?

**Describe the solution you'd like**
Add a column, `endpoint_applications` or similar, to show the list of applications installed on the device.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Subhajit97**
# Example query results
<details>
  <summary>Results</summary>

```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: docs/tables/vanta_computer.md [hunk 1]
```diff
@@ -101,6 +101,64 @@ where
...
+  endpoint_applications is not null
+  and id not in (
+    select
+      id
+    from
+      device_with_slack_installed
+  );
+```
+
+### List computers with older version of Zoom (< 5.12)
```
Comment (misraved) [inline docs/tables/vanta_computer.md:146]:
```suggestion
### List computers with an older version of Zoom app (< 5.12)
```

File: vanta/table_vanta_computer.go [hunk 2]
```diff
@@ -76,3 +80,47 @@ func listVantaComputers(ctx context.Context, d *plugin.QueryData, _ *plugin.Hydr
...
+	// Return nil, if the device is unmonitored / unsupported.
+	var unsupportedReasons api.WorkstationUnsupportedReasons
+	if computer.UnsupportedReasons != unsupportedReasons {
+		return nil, nil
+	}
+
+	// Create client
+	conn, err := getVantaAppClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("vanta_computer.listVantaComputers", "connection_error", err)
```
Comment (misraved) [inline vanta/table_vanta_computer.go:101]:
```suggestion
		plugin.Logger(ctx).Error("vanta_computer. listVantaComputerApplications", "connection_error", err)
```

**Review by misraved — COMMENTED** [review]
@Subhajit97 please take a look at the review comments/questions. Thanks!!

File: vanta/table_vanta_computer.go [hunk 3]
```diff
@@ -76,3 +80,47 @@ func listVantaComputers(ctx context.Context, d *plugin.QueryData, _ *plugin.Hydr
...
+		// Remove all entries with empty data
+		if strings.ReplaceAll(app, " ", "") == "" {
+			continue
+		}
+
+		// Remove all duplicate data
+		if !helpers.StringSliceContains(result, app) {
+			result = append(result, app)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Subhajit97) [reply vanta/table_vanta_computer.go:123]: @misraved This behaviour is not related to the API; even the UI shows empty bullet points when we try to list the applications. Probably those have some privacy policy enabled on them.
Screenshot 2023-02-09 at 12 52 12 PM

For the duplicate item, the UI also shows these data. It is possible if the same version of an app has been installed more than once using different installation methods.

**Review by misraved — APPROVED** [review]

File: vanta/table_vanta_computer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (misraved) [reply vanta/table_vanta_computer.go:123]: Thanks @Subhajit97 for the confirmation. It does look weird with the empty and duplicate data, so it is better to remove them for now 👍.

**Pull request merged**